File: tests/.hooks/php/custom-http/src/Hooks/HookRegistration.php
<?php

declare(strict_types=1);

namespace OpenAPI\OpenAPI\Hooks;

class HookRegistration
{
    /**
     * @param  Hooks  $hooks
     */
    public static function initHooks(Hooks $hooks): void
    {
        $customSecurityHook = new CustomSecurityHook();
        $hooks->registerBeforeRequestHook($customSecurityHook);
    }
}

File: zSDKs/sdk-csharp/src/Speakeasy/OpenAPI/Hooks/HookRegistration.cs
namespace Speakeasy.OpenAPI
{
    /// <summary>
    /// This file was modified and should not get overridden.
    /// </summary>
    public static class HookRegistration
    {
        public static void InitHooks(IHooks hooks)
        {
            hooks.RegisterBeforeRequestHook(new IdempotencyHook());
        }
    }
}
